File: src/util/GlobalFunction.js
import { record } from "./prompt.js";

export const validNumber = (
  numberValue,
  limit = null,
  fieldName,
  objetName
) => {
  const number = Number(numberValue);
  if (
    !numberValue ||
    numberValue < 0 ||
    isNaN(number) ||
    ((limit || limit == 0) && limit < numberValue)
  ) {
    throw new Error(
      `Field "${fieldName || ""}" ${
        objetName ? "in object named" + objetName : ""
      } is invalid. `
    );
  }

  return number;
};

export const getNumberValue = async (unit) => {
  return await record(`Enter a number value (${unit}): `);
};

File: util/globalFunction.dart
import 'dart:async';

import './prompt.dart'; // Import the record function from prompt.dart

Future<double> getNumberValue(String unit) async {
  final input = await record('Enter a number value ($unit): ');
  final number = double.tryParse(input);
  if (number == null) {
    throw Exception('Invalid number entered.');
  }
  return number;
}


bool validNumber(
  double? numberValue, {
  double? limit,
  String? fieldName,
  String? objectName,
}) {
  if (numberValue == null ||
      numberValue < 0 ||
      numberValue.isNaN ||
      (limit != null && numberValue > limit)) {
    throw Exception(
      'Field "$fieldName" ${objectName != null ? "in object named $objectName" : ""}'
      ' is invalid.',
    );
  }
  return true;
}
